.jenkinsci-new/builders/x64-linux-build-steps.groovy

```diff
@@ -166,8 +170,23 @@ def successPostSteps(scmVars, boolean packagePush, String dockerTag, List enviro
   }
 }
 
-def alwaysPostSteps(List environment) {
+def alwaysPostSteps(scmVars, List environment, boolean coredumps) {
   stage('Linux always PostSteps') {
+    // handling coredumps (if tests crashed)
+    if (currentBuild.currentResult != "SUCCESS" && coredumps) {
+      def dumpsFileName = sprintf('coredumps-%1$s.bzip2',
+        [scmVars.GIT_COMMIT.substring(0,8)])
+
+      sh(script: "echo 'build/bin' > coredumps.upload")
+      sh(script: "find . -type f -name '*.coredump' -exec echo '{}' \\; >> coredumps.upload")
+      sh(script: "tar -cjvf ${dumpsFileName} -T coredumps.upload")
+      if( fileExists(dumpsFileName)) {
+        withCredentials([usernamePassword(credentialsId: 'ci_nexus', passwordVariable: 'NEXUS_PASS', usernameVariable: 'NEXUS_USER')]) {
+          sh(script: "curl -u ${NEXUS_USER}:${NEXUS_PASS} --upload-file ${WORKSPACE}/${dumpsFileName} https://nexus.iroha.tech/repository/artifacts/iroha/coredumps/${dumpsFileName}")
+        }
+        echo "Build is not SUCCESS! See core dumps at: https://nexus.iroha.tech/service/rest/repository/browse/artifacts/iroha/coredumps/${dumpsFileName}"
```

Comment:
I tested:
```
Build is not SUCCESS! See core dumps at: https://nexus.iroha.tech/service/rest/repository/browse/artifacts/iroha/coredumps/coredumps-29af969e.bzip2
```
Link dose not work,
but artifact is uploaded:
https://nexus.iroha.tech/repository/artifacts/iroha/coredumps/coredumps-29af969e.bzip2

Reply:
@bakhtin, may be use web UI link?
https://nexus.iroha.tech/#browse/browse:artifacts:iroha/coredumps/coredumps-5ea47c5f.bzip2

Reply:
I provided a working link above. It should be `https://nexus.iroha.tech/service/rest/repository/browse/artifacts/iroha/coredumps/`

Reply:
@bakhtin so the message should look like this? :
```
Build is not SUCCESS! See core dumps "coredumps-29af969e.bzip2" at: https://nexus.iroha.tech/service/rest/repository/browse/artifacts/iroha/coredumps/
```
because in you url no coredump name is  present.

Reply:
ofc, the name of the coredump itself should be appended. I just provided a base URL

Reply:
https://nexus.iroha.tech/service/rest/repository/browse/artifacts/iroha/coredumps/ - link with folders
https://nexus.iroha.tech/repository/artifacts/iroha/coredumps/coredumps-f4f27dc2.bzip2 - link with file.

This paths are different, I used direct link to easy "click & download"